### PR TITLE
Remote backend

### PR DESCRIPTION
**Pull request by jansol**
Adds a new "PoCL-Remote" backend for offloading OpenCL commands across a network, featuring smart buffer management and distributed command scheduling.

### CODE REVIEW

File: doc/sphinx/source/remote.rst
Comment (pjaaskel) [inline doc/sphinx/source/remote.rst:26]: Perhaps advertise the aspect that we can build a Nano-PoCL with low requirements on the target system and compiler, and point to the nanodrone case. This is why the client-side driver and the core PoCL is written in simple C instead of C++, for example.

File: lib/CL/pocl_lttng.h
Comment (jansol) [reply lib/CL/pocl_lttng.h:45]: That's what clang-format insists on for this. Reformatted manually.

**Review by pjaaskel — APPROVED** [review]
LGTM after the tests pass. Do the failures look relevant?

**Comment by jansol** [comment]
Remaining CI failures are:
* some invalid pointer assignment in pocl-vulkan
* examples/matrix1_spir_local printing a different result than the expected one
* ubsan disliking opencl.hpp (I thought this was fixed in the latest version of that header but apparently not)

File: CMakeLists.txt
Comment (franz) [inline CMakeLists.txt:890]: This (SINGLE_LLVM_LIB CMake option) was removed in a0488d6f152670f5de, and STATIC_LLVM was reintroduced.

**Comment by franz** [comment]
WRT UBsan + opencl.hpp, with main branch it works *if* the C++ standard is set high enough. The github runner uses `-DENABLE_LATEST_CXX_STD=ON`, which sets the default C++ standard to 20. IIRC with C++17 it compiles *unless* using UBsan, and with older C++ standards it doesn't compile at all. But I could be wrong, it's been a while since i touched UBsan.

File: CMakeLists.txt
Comment (jansol) [reply CMakeLists.txt:890]: re-removed the block

**Comment by jansol** [comment]
> WRT UBsan + opencl.hpp, with main branch it works _if_ the C++ standard is set high enough. The github runner uses `-DENABLE_LATEST_CXX_STD=ON`, which sets the default C++ standard to 20. IIRC with C++17 it compiles _unless_ using UBsan, and with older C++ standards it doesn't compile at all. But I could be wrong, it's been a while since i touched UBsan.

Could it be that the GCC 10 on the runner is just too old for this opencl.hpp then? UBsan builds are fine with my local GCC 12 and I think the opencl.hpp on main is quite a bit older.

File: CMakeLists.txt
Comment (isuruf) [inline CMakeLists.txt:91]:
```suggestion
    set(FORBID_IMPLICIT_FUNCTIONS "-Werror=implicit-function-declaration")
```

File: CMakeLists.txt
Comment (isuruf) [inline CMakeLists.txt:93]:
```suggestion
      set(FORBID_IMPLICIT_FUNCTIONS ${FORBIT_IMPLICIT_FUNCTIONS} "-Werror=incompatible-pointer-types")
```

File: doc/www/index.mak
Comment (isuruf) [inline doc/www/index.mak:29]:
```suggestion
memory management and distributed command scheduling to keep latency down
```

**Comment by pjaaskel** [comment]
Updated the web and ran the tests tested locally successfully (`ctest -R remote`). Good to go from my side, as soon as @franz approves.

**Comment by jansol** [comment]
Filled in the TODO arXiv links in docs.

**Pull request merged**
